### PR TITLE
dbus: 1.12.20 -> 1.14.0

### PR DESCRIPTION
**Pull request by Artturin**
https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.14.0/NEWS

the reproducibility patch is no longer needed,  also, it didn't work anymore anyways

- Built on platform(s)
  - [x] x86_64-linux and cross aarch64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/dbus/default.nix [hunk 1]
```diff
@@ -8,42 +8,26 @@
...
+    url = "https://dbus.freedesktop.org/releases/dbus/dbus-${version}.tar.xz";
+    sha256 = "sha256-zNfM43WW4KGVWP1mSNEnKrQ/AR2AyGNa6o/QutWK69Q=";
   };
 
   patches = [
-    # 'generate.consistent.ids=1' ensures reproducible docs, for further details see
-    # http://docbook.sourceforge.net/release/xsl/current/doc/html/generate.consistent.ids.html
-    # Also applied upstream in https://gitlab.freedesktop.org/dbus/dbus/-/merge_requests/189,
-    # expected in version 1.14
-    ./docs-reproducible-ids.patch
```
Comment (jtojnar) [inline pkgs/development/libraries/dbus/default.nix:37]: Remove this file as well.

**Comment by jtojnar** [comment]
Also need to remove dbus patch from `fprintd`.

File: pkgs/development/libraries/dbus/default.nix [hunk 2]
```diff
@@ -68,25 +52,26 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" "lib" "doc" "man" ];
 
+  strictDpes = true;
```
Comment (jtojnar) [inline pkgs/development/libraries/dbus/default.nix:55]: `strictDpes` → `strictDeps`

File: pkgs/development/libraries/dbus/default.nix [hunk 3]
```diff
@@ -68,25 +52,26 @@ stdenv.mkDerivation rec {
...
   nativeBuildInputs = [
     autoreconfHook
     autoconf-archive
     pkg-config
     docbook_xml_dtd_44
     docbook-xsl-nons
     xmlto
-  ];
+  ] ++ lib.optional enableSystemd systemd;
```
Comment (jtojnar) [inline pkgs/development/libraries/dbus/default.nix:63]: Won’t this cause build systemd to be used instead of host one in the service file?

Comment (Artturin) [reply pkgs/development/libraries/dbus/default.nix]: yeah, i now defined it manually like the docs say https://cgit.freedesktop.org/dbus/dbus/tree/configure.ac#n1027

File: pkgs/development/libraries/dbus/default.nix [hunk 4]
```diff
@@ -82,15 +67,16 @@ stdenv.mkDerivation rec {
...
       libX11
       libICE
       libSM
-    ] ++ lib.optional enableSystemd systemd
+    ]) ++ lib.optional enableSystemd systemd
     ++ lib.optionals stdenv.isLinux [ audit libapparmor ];
   # ToDo: optional selinux?
 
   configureFlags = [
+    "SYSTEMCTL=${systemd}/bin/systemctl"
```
Comment (Mindavi) [inline pkgs/development/libraries/dbus/default.nix:79]: Should be optional on `enableSystemd`

Comment (SuperSandro2000) [reply pkgs/development/libraries/dbus/default.nix:79]: Should fix the eval issue.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by jtojnar** [comment]
Looks like the catalog paths are broken, see https://github.com/NixOS/nixpkgs/pull/174043

**Pull request merged**